### PR TITLE
Export configured Vite plugins from `packages/build`

### PR DESCRIPTION
**Pull request by gzdunek**
Currently, both Web UI and Connect import plugin configs from `packages/build` and then provide them to plugins they import. This will no longer work after the migration to pnpm, because the plugins are installed in `packages/build`, so they are not available outside that package.

Instead of moving them to the root, we can export configured plugins from `packages/build`, reducing the code duplication.
Unfortunately, it turned out to be more difficult than I thought:
1. Importing the plugins via relative paths (`import { reactPlugin } from '../build/vite/react'`) didn't work, because the plugin packages (like `@vitejs/plugin-react-swc`) were resolved in context of a final package (like teleterm or teleport) so Node.js couldn't find them in node_modules.
2. Using workspace imports fixed that issue, but I couldn't make it work with TypeScript. I was getting errors as below: `TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /Users/grzegorz/code/teleport/web/packages/build/vite/tsconfigPaths.ts`
I tried different options, importing with no extension, with `.mts` etc, but with no luck.
AFAIK the only way to solve it is to use `ts-node` or something to allow Node.js resolve TS imports. However, this is too much work. Because our plugins are very simple, I decided to convert them to JS, I don't think we will lose much on this.
There is [a long issue about this problem in Vite](https://github.com/vitejs/vite/issues/5370).

### CODE REVIEW

**Review by ravicious — APPROVED** [review]

**Comment by gzdunek** [comment]
Friendly ping @ryanclark

File: web/packages/build/vite/react.mjs [hunk 1]
```diff
@@ -16,7 +16,23 @@
...
+/**
+ * This needs to be an .mjs file,
+ * because Vite can't import .ts files from a workspace out of the box.
+ * https://github.com/vitejs/vite/issues/5370
+ */
+
+import react from '@vitejs/plugin-react-swc';
+
+/** @param {string} mode */
+export const reactPlugin = mode => {
```
Comment (ryanclark) [inline web/packages/build/vite/react.mjs:28]: Can this be a function to be consistent with the others

**Review by ryanclark — COMMENTED** [review]
What's the motivation for removing TypeScript? Can we use `.mts` instead?

**Comment by gzdunek** [comment]
>What's the motivation for removing TypeScript? Can we use .mts instead?

I wasn't able to import TS files from a `build` workspace (please take a look at the point 2. in the PR description). I tried `.mts` too, but with no luck, I think we would have to use a custom module loader (like `ts-node`) to be able to load a TS file :/
This should be probably fixed on Vite side, since it can load TS files imported through relative imports.

**Comment by ryanclark** [comment]
> (please take a look at the point 2. in the PR description)

Yep, my bad

**Review by ryanclark — APPROVED** [review]

**Pull request merged**